lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -63,6 +50,31 @@
     cert_validation_mode='AZURE_CERT_VALIDATION_MODE',
 )
 
+# FUTURE: this should come from the SDK or an external location.
+# For now, we have to copy from azure-cli
+AZURE_API_PROFILES = {
+    'latest': {
+        'ContainerInstanceManagementClient': '2017-10-01-preview',
```

Comment:
ContainerInstanceManagementClient doesn't support multi-api profile yet.

Reply:
So, btw, just a related question. Shouldn't single-api modules still accept api_version parameter, but just ignore it? I was doing some tests and tried to specify current api version for single-api module just in case newer api appears and this fails because of invalid parameter.
Of course this could be somehow handled, but would be much easier if this parameter was always defined.

Reply:
@lmazuel yeah, this is what worries me about the unversioned clients- if tomorrow a new version of containermanagement came out and the SDK was updated to multi-version, we'd have no way to communicate the profile to it (to keep it on the version we tested with) without using inspect and passing api_profile only if it's supported (as this PR does). That's why I've added ContainerInstanceManagementClient to the profile- so that if/when it *is* supported, we stay on the REST API version we tested rather than floating to the new current (which is what would happen if we don't pass profile, right?). So it seems like some version of this hack will be necessary unless every client accepts profile/api_version regardless of current single/multi-version status...